batch/batch/worker/worker.py

```diff
@@ -3048,13 +3048,41 @@ async def create_job_1(self, request):
 
         batch_id = body['batch_id']
         job_id = body['job_id']
+        id = (batch_id, job_id)
 
         format_version = BatchFormatVersion(body['format_version'])
 
         token = body['token']
         start_job_id = body['start_job_id']
         addtl_spec = body['job_spec']
 
+        user = body['user']
+        gsa_key = body['gsa_key']
+
+        # already running
+        if id in self.jobs:
+            return web.HTTPForbidden()
+
+        # check worker hasn't started shutting down
+        if not self.active:
+            return web.HTTPServiceUnavailable()
+
+        request['batch_telemetry'] = {
+            'operation': 'create_job',
+            'batch_id': str(batch_id),
+            'job_id': str(job_id),
+            'job_queue_time': str(body['queue_time']),
+        }
+
+        self.task_manager.ensure_future(
+            self.create_job_2(
+                batch_id, job_id, format_version, token, start_job_id, addtl_spec, user, credentials=gsa_key
+            )
+        )
+
+        return web.Response()
+
+    async def create_job_2(self, batch_id, job_id, format_version, token, start_job_id, addtl_spec, user, credentials):
```

Comment:
How do I make sure that the job is marked with an error if there's a problem before the `run_job` call?

Reply:
You'll have to cook up a call to `Worker.post_job_complete_1` (might need to refactor that since you might not have a `Job` object) instead of calling `run_job`.